src/Filter/ScssFilter.php

```diff
@@ -51,4 +54,130 @@ public function input($filename, $input)
         $return = $this->_runCmd($bin, '', array('PATH' => $this->_settings['path']));
         return $return;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getDependencies(AssetTarget $target)
+    {
+        $children = [];
+        foreach ($target->files() as $file) {
+            $imports = CssUtils::extractImports($file->contents());
+            if (empty($imports)) {
+                continue;
+            }
+
+            $ext = $this->_settings['ext'];
+            $extLength = strlen($ext);
+
+            $deps = [];
+            foreach ($imports as $name) {
+                if (preg_match('/(.*)\/\*\*\/\*$/', $name, $matches)) {
+                    $path = $this->_findFile($matches[1]);
+                    $relPathBegin = mb_strlen($path) - mb_strlen($matches[1]);
+                    $result = [];
+                    $this->_dirToArray($path, $relPathBegin, $result);
+                    $deps = array_merge($deps, $result);
+                    continue;
+                }
+                $pathinfo = pathinfo($name);
+                if (substr($pathinfo['basename'], 0, 1) !== '_') {
+                    $name = $pathinfo['dirname'] . '/_' . $pathinfo['basename'];
+                }
+                if ($ext !== substr($name, -$extLength)) {
+                    $name .= $ext;
+                }
+                $deps[] = $name;
+            }
+            foreach ($deps as $import) {
+                $path = $this->_findFile($import);
+                try {
+                    $file = new Local($path);
+                    $newTarget = new AssetTarget('phony.css', [$file]);
+                    $children[] = $file;
+                } catch (\Exception $e) {
+                    // Do nothing, we just skip missing files.
+                    // sometimes these are things like compass or bourbon
+                    // internals.
+                    $newTarget = false;
+                }
+
+                // Only recurse through non-css imports as css files are not
+                // inlined by less/sass.
+                if ($newTarget && $ext === substr($import, -$extLength)) {
+                    $children = array_merge($children, $this->getDependencies($newTarget));
+                }
+            }
+        }
+        return $children;
+    }
+
+    /**
+     * Discover all the sub directories for a given path.
+     *
+     * @param string $path The path to search
+     * @return array Array of subdirectories.
+     */
+    protected function _generateTree($path)
+    {
+        $paths = glob($path, GLOB_ONLYDIR);
+        if (!$paths) {
+            $paths = array();
+        }
+        $basepath = dirname($path);
+        if (pathinfo($basepath, PATHINFO_BASENAME) === '**') {
+            $paths = array_merge($paths, glob($basepath, GLOB_ONLYDIR));
+            $basepath = dirname($basepath);
+        }
+        array_unshift($paths, $basepath);
+        return $paths;
```

Comment:
We surely need expanding the ** operator in AssetScanner's path expansion. And we need the same for ScssFilter as well. That's because ScssFilter needs to be aware of its dependency scss files. Getting dependencies vary from filter to filter, doesn't it? AssetScanner's path expansion is needed for locating build file paths.